[9/n] [torch/elastic] Introduce RendezvousSettings

cbalioglu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57151 [23/n] [torch/elastic] Introduce the implementation of DynamicRendezvousHandler
* #57150 [22/n] [torch/elastic] Introduce a new from_backend static constructor for DynamicRendezvousHandler
* #57149 [21/n] [torch/elastic] Introduce _RendezvousJoinOp
* #57148 [20/n] [torch/elastic] Introduce _RendezvousExitOp
* #57147 [19/n] [torch/elastic] Introduce _RendezvousKeepAliveOp
* #57146 [18/n] [torch/elastic] Introduce _RendezvousCloseOp
* #57145 [17/n] [torch/elastic] Introduce _DistributedRendezvousOpExecutor
* #57144 [16/n] [torch/elastic] Introduce _RendezvousOpExecutor
* #56538 [15/n] [torch/elastic] Introduce _RendezvousStateHolder
* #57143 [14/n] [torch/elastic] Introduce a name attribute to _PeriodicTimer
* #57142 [13/n] [torch/elastic] Extend the return type of RendezvousBackend's set_state method
* #57141 [12/n] [torch/elastic] Rename last_keep_alives to last_heartbeats in _RendezvousState
* #57140 [11/n] [torch/elastic] Add heartbeat timeout to RendezvousTimeout
* #57139 [10/n] [torch/elastic] Add comparison operators to _NodeDesc
* **#56537 [9/n] [torch/elastic] Introduce RendezvousSettings**
* #56536 [8/n] [torch/elastic] Add unit tests for _RendezvousState
* #56535 [7/n] [torch/elastic] Rename _Rendezvous to _RendezvousState
* #56534 [6/n] [torch/elastic] Reorder type definitions in dynamic_rendezvous.py
* #56533 [5/n] [torch/elastic] Introduce the delay utility function
* #56532 [4/n] [torch/elastic] Fix the finalizer of PeriodicTimer

This PR introduces the `RendezvousSettings` type to consolidate the arguments passed to `DynamicRendezvousHandler`.

Differential Revision: [D27890155](https://our.internmc.facebook.com/intern/diff/D27890155/)

cbalioglu committed:
[9/n] [torch/elastic] Introduce RendezvousSettings

This PR introduces the `RendezvousSettings` type to consolidate the arguments passed to `DynamicRendezvousHandler`.

Differential Revision: [D27890155](https://our.internmc.facebook.com/intern/diff/D27890155/)

[ghstack-poisoned]

cbalioglu committed:
Update on "[9/n] [torch/elastic] Introduce RendezvousSettings"


This PR introduces the `RendezvousSettings` type to consolidate the arguments passed to `DynamicRendezvousHandler`.

Differential Revision: [D27890155](https://our.internmc.facebook.com/intern/diff/D27890155/)

[ghstack-poisoned]

tierex commented on torch/distributed/elastic/rendezvous/dynamic_rendezvous.py:
```diff
@@ -140,6 +140,18 @@ def _set_timeouts(self, **timeouts: Optional[timedelta]):
 
 
+@dataclass(repr=False, eq=False, frozen=True)
+class RendezvousSettings:
```
Why can't we reuse https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/rendezvous/api.py#L124 ?

cbalioglu replied:
First of all, unlike `RendezvousParameters`, `RendezvousSettings` is immutable/read-only. Beyond that `RendezvousSettings` does not map one-to-one with rendezvous parameters.  For instance the endpoint and the backend name are not part of it. However it has two additional attributes related to keep-alive settings (which are constants for now) and in the future it will hold other attributes that will be derived from a combination of passed parameters and the environment in which the rendezvous runs.

cbalioglu committed:
Update on "[9/n] [torch/elastic] Introduce RendezvousSettings"


This PR introduces the `RendezvousSettings` type to consolidate the arguments passed to `DynamicRendezvousHandler`.

Differential Revision: [D27890155](https://our.internmc.facebook.com/intern/diff/D27890155/)

[ghstack-poisoned]

cbalioglu committed:
Update on "[9/n] [torch/elastic] Introduce RendezvousSettings"


This PR introduces the `RendezvousSettings` type to consolidate the arguments passed to `DynamicRendezvousHandler`.

Differential Revision: [D27890155](https://our.internmc.facebook.com/intern/diff/D27890155/)

[ghstack-poisoned]
